src/box/lua/config/applier/console.lua

```diff
@@ -28,6 +31,18 @@ local function apply(config)
     local iconfig_def = configdata._iconfig_def
     socket_file = instance_config:prepare_file_path(iconfig_def, socket_file)
 
+    if old_socket_file == nil then
+        old_socket_file = socket_file
+    end
+
+    if old_socket ~= nil and socket_file ~= old_socket_file then
+        if type(old_socket.shutdown) == 'function' then
+            old_socket:shutdown()
+        end
```

Comment:
`shutdown` and `close` are not interchangable. You should call `close` after shutdown anyway, since only `close` actually destroys the socket.